### ISSUE TITLE
Heimgard Smart Lock

### ISSUE THREAD

**Issue by donnib** [dresden-elektronik/deconz-rest-plugin#8073]
### Is there already an existing issue for this?

- [X] I have searched the existing issues and there is none for my device

### Product name

Heimgard Smart Lock

### Manufacturer

Heimgard Technologies

### Model identifier

HT-SLM-2

### Device type to add

Lock

### Node info

<img width="510" alt="image" src="https://github.com/user-attachments/assets/05b6cff6-f17d-47e6-b584-d4c0b0f286eb" />

### Endpoints and clusters

<img width="342" alt="image" src="https://github.com/user-attachments/assets/7b153972-abb1-4fab-ae5e-9cf235ea2644" />

### Basic

<img width="1038" alt="image" src="https://github.com/user-attachments/assets/797b990b-36b2-4e4c-a2ec-4d4c640fa6a7" />

### Further relevant clusters

<img width="515" alt="image" src="https://github.com/user-attachments/assets/0c470954-65b2-4699-9c4c-69a7b8bae52c" />

<img width="510" alt="image" src="https://github.com/user-attachments/assets/7a356cd6-352b-4f44-a808-2d8251d3b7d5" />

<img width="505" alt="image" src="https://github.com/user-attachments/assets/bceba77e-884e-47d8-8c5d-e9c589b40021" />

<img width="507" alt="image" src="https://github.com/user-attachments/assets/0f9be0b1-f14f-4d0e-9fa7-512c9668bc59" />

<img width="483" alt="image" src="https://github.com/user-attachments/assets/490726f3-0764-4969-b2c9-950781cb9e1c" />

<img width="484" alt="image" src="https://github.com/user-attachments/assets/01552f93-cace-4ebe-8271-3dcae7b11d80" />

<img width="476" alt="image" src="https://github.com/user-attachments/assets/374fed81-0709-4fbb-8453-bcb19c1d8bb7" />

<img width="483" alt="image" src="https://github.com/user-attachments/assets/190e1c01-fcb1-462a-b44d-3e5e1e967794" />

<img width="487" alt="image" src="https://github.com/user-attachments/assets/195de543-0bc5-49bd-934e-f748e7a94aa6" />

<img width="494" alt="image" src="https://github.com/user-attachments/assets/8994e20a-fab0-48a2-894a-94bb236c52db" />

**Comment by Smanar** [comment]
Hello can try this DDF 

```
{
  "schema": "devcap1.schema.json",
  "manufacturername": "Heimgard Technologies",
  "modelid": "HT-SLM-2",
  "vendor": "Heimgard",
  "product": "Door lock (HC-SLM-2)",
  "sleeper": false,
  "status": "Gold",
  "subdevices": [
    {
      "type": "$TYPE_DOOR_LOCK_SENSOR",
      "restapi": "/sensors",
      "uuid": [
        "$address.ext",
        "0x01",
        "0x0101"
      ],
      "items": [
        {
          "name": "attr/id"
        },
        {
          "name": "attr/lastannounced"
        },
        {
          "name": "attr/lastseen"
        },
        {
          "name": "attr/manufacturername"
        },
        {
          "name": "attr/modelid"
        },
        {
          "name": "attr/name"
        },
        {
          "name": "attr/swversion"
        },
        {
          "name": "attr/type"
        },
        {
          "name": "attr/uniqueid"
        },
        {
          "name": "config/battery",
          "read": {
            "at": "0x0021",
            "cl": "0x0001",
            "ep": 1,
            "fn": "zcl:attr"
          },
          "parse": {
            "at": "0x0021",
            "cl": "0x0001",
            "ep": 1,
            "eval": "Item.val = Attr.val/2",
            "fn": "zcl:attr"
          },
          "default": 0
        },
        {
          "name": "config/lock",
          "read": {
            "at": "0x0000",
            "cl": "0x0101",
            "ep": 1,
            "fn": "zcl:attr"
          },
          "parse": {
            "at": "0x0000",
            "cl": "0x0101",
            "ep": 1,
            "eval": "Item.val = Attr.val ==1",
            "fn": "zcl:attr"
          }
        },
        {
          "name": "config/reachable"
        },
        {
          "name": "state/lockstate",
          "read": {
            "at": "0x0000",
            "cl": "0x0101",
            "ep": 1,
            "fn": "zcl:attr"
          },
          "parse": {
            "at": "0x0000",
            "cl": "0x0101",
            "ep": 1,
            "eval": "if (Attr.val == 0) { Item.val = 'not fully locked' } else if (Attr.val == 1) { Item.val = 'locked' } else if (Attr.val == 2) { Item.val = 'unlocked' } else { Item.val = 'undefined' }",
            "fn": "zcl:attr"
          }
        },
        {
          "name": "state/open",
          "read": {
            "at": "0x0003",
            "cl": "0x0101",
            "ep": 1,
            "fn": "zcl:attr"
          },
          "parse": {
            "at": "0x0003",
            "cl": "0x0101",
            "ep": 1,
            "eval": "Item.val = Attr.val !=1",
            "fn": "zcl:attr"
          }
        }
      ]
    }
  ],
  "bindings": [
    {
      "bind": "unicast",
      "src.ep": 1,
      "cl": "0x0101",
      "report": [
        {
          "at": "0x0000",
          "dt": "0x30",
          "min": 5,
          "max": 300
        },
        {
          "at": "0x0003",
          "dt": "0x30",
          "min": 5,
          "max": 300
        }
      ]
    },
    {
      "bind": "unicast",
      "src.ep": 1,
      "cl": "0x0001",
      "report": [
        {
          "at": "0x0021",
          "dt": "0x20",
          "min": 300,
          "max": 2700,
          "change": "0x00000001"
        }
      ]
    }
  ]
}
```

Have mimic an actual one, not sure for battery as on your capture both lines are at "0".

**Comment by donnib** [comment]
Thank you, i can't load the file. I tried following :

1. Make a new file on the rpi with the contents above and save it as json
2. Open the node in deconz ui, edit DDF
3. Click File -> Open and choose the json file from step 1
-> Failed to open 

What am i doing wrong ?

**Comment by donnib** [comment]
<img width="1209" alt="image" src="https://github.com/user-attachments/assets/6b8f5e4e-aeff-4cc7-81fc-53cbef1b64c9" />

**Comment by Smanar** [comment]
What is you  OS ?
The json need to be in a "devices" folder, then just restart deconz, never use the editor, except to test/check.

> 
All DDF provided directly with deCONZ typically reside in /usr/share/deCONZ/devices/ on a Linux system and are loaded first. However, files residing in the home directory of the user running deCONZ (e.g. /home/<DECONZUSER>/.local/share/dresden-elektronik/deCONZ/devices) will override the pre-packaged files to allow users to amend and keep their own files if desired

**Comment by donnib** [comment]
I am running deconz on a rPi and i believe its a Debian distribution. I wanted to test the file first before restarting second. Why is that not possible ?

**Comment by Smanar** [comment]
Deconz load DDF only at start and too much problem using the editor.

**Comment by donnib** [comment]
I have a folder called : /.local/share/data/dresden-elektronik/deCONZ/devices
should i call the file a specific name because i tried making a file called lock.json and place the content above and restarted deconz but nothing happened to the lock, do i need to repair the lock again or ?

**Comment by Smanar** [comment]
No, it just need to be in the folder "devices", can make a subfolder caller Heimgard.
And yes you need to pait it again, and this time deconz will use the DDF to reconise it, if you edit the DDF later it will be not necessary, but it is for the first inclusion (ATM the device is not reconised, so not in the API at all)

**Comment by donnib** [comment]
Got it, i repaired and now it shows up however i was hoping i would get events from the lock when it's unlocked like which code was used to unlock to identify the person but i don't see any events in HA either because the lock doesn't send them or something else is needed more in the DFF ? (i don't know if DFF files have any implications on this)

**Comment by Smanar** [comment]
Nope the RFID (or code gestion) need this kind of PR https://github.com/dresden-elektronik/deconz-rest-plugin/pull/5918
The working mode ATM is realy basic.
